scripts/data_contamination/load_documents.py

```diff
@@ -0,0 +1,139 @@
+import json
+
+from typing import Iterator
+
+
+def get_document_iterator(file_path: str, file_format: str) -> Iterator[str]:
+    if file_format in ["the_pile", "oig", "c4"]:
+        return get_the_pile_document_iterator(file_path)
+    elif file_format == "flan":
+        return get_flan_document_iterator(file_path)
+    elif file_format in ["xp3", "p3"]:
+        return get_xp3_document_iterator(file_path)
+    elif file_format == "root":
+        return get_root_document_iterator(file_path)
+    elif file_format == "natural":
+        return get_natural_document_iterator(file_path)
+    elif file_format == "hh":
+        return get_root_document_iterator(file_path)
+    elif file_format == "raw":
+        return get_raw_document_iterator(file_path)
+    elif file_format == "custom":
+        return get_custom_document_iterator(file_path)
+    else:
+        raise NotImplementedError()
+
+
+def get_hh_document_iterator(file_path: str) -> Iterator[str]:
+    """
+    This method reads input files with similar file formats with Anthropic/hh-rlhh jsonl format.
+
+    example:
+    {'chosen': '\n\nHuman: What kind of noises did dinosaurs make?\n\nAssistant: Humans and dinosaurs didn’t live at the same time, so it’s really hard to say. The best place to find out what noises dinosaurs made would be\n\nHuman: yes they did\n\nAssistant: to guess, and that would probably require lots of reading and a certain amount of imagination, so we’re not really prepared to do that.\n\nHuman: you cant read\n\nAssistant: You can read?',
+    'rejected': '\n\nHuman: What kind of noises did dinosaurs make?\n\nAssistant: Humans and dinosaurs didn’t live at the same time, so it’s really hard to say. The best place to find out what noises dinosaurs made would be\n\nHuman: yes they did\n\nAssistant: to guess, and that would probably require lots of reading and a certain amount of imagination, so we’re not really prepared to do that.\n\nHuman: you cant read\n\nAssistant: there’s a lot of stuff humans don’t know'}
+
+    """
+    with open(file_path, "r") as f:
```

Comment:
For all the following we're just treating all field types as the same as POC; should add in granularity when running for real. Same with other parsers